### PR TITLE
[5.1] Fix bug in Pagination when perPage is less than zero

### PR DESCRIPTION
**Pull request by phpawy**
Show a user friendly error message when you pass perPage parameter less than zero to function paginate()

example code for reproduce error:
<?php
.....
$adsnum = 25;
$output = Ads::paginate(15 - $adsnum);
?>

### CODE REVIEW

File: src/Illuminate/Database/Eloquent/Builder.php [hunk 1]
```diff
@@ -268,6 +269,10 @@ public function lists($column, $key = null)
      */
     public function paginate($perPage = null, $columns = ['*'], $pageName = 'page', $page = null)
     {
+        if ($perPage < 0) {
+            throw new RuntimeException("Invalid argument perpage.");
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Builder.php]: Why did you change the exception message?

File: src/Illuminate/Database/Eloquent/Relations/BelongsToMany.php [hunk 2]
```diff
@@ -197,6 +198,9 @@ public function get($columns = ['*'])
      */
     public function paginate($perPage = null, $columns = ['*'], $pageName = 'page')
     {
+        if ($perPage < 0) {
+            throw new RuntimeException("Invalid argument perpage.");
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Relations/BelongsToMany.php]: What?

Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Relations/BelongsToMany.php]: You need to throw an InvalidArgumentException.

File: src/Illuminate/Database/Eloquent/Relations/BelongsToMany.php [hunk 3]
```diff
@@ -197,6 +198,9 @@ public function get($columns = ['*'])
      */
     public function paginate($perPage = null, $columns = ['*'], $pageName = 'page')
     {
+        if ($perPage < 0) {
+            throw new RuntimeException("Invalid argument perpage."); 
+        }
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Relations/BelongsToMany.php]: missing newline

File: src/Illuminate/Database/Query/Builder.php [hunk 4]
```diff
@@ -1399,6 +1400,9 @@ protected function runSelect()
      */
     public function paginate($perPage = 15, $columns = ['*'], $pageName = 'page')
     {
+        if ($perPage < 0) {
+            throw new RuntimeException("Invalid argument perpage."); 
+        }
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Builder.php]: missing newline

**Comment by phpawy** [comment]
fixed, could you please review now?

File: src/Illuminate/Database/Eloquent/Builder.php [hunk 5]
```diff
@@ -268,6 +269,10 @@ public function lists($column, $key = null)
      */
     public function paginate($perPage = null, $columns = ['*'], $pageName = 'page', $page = null)
     {
+        if ($perPage < 0) {
+            throw new InvalidArgumentException('$perPage cannot be a negative value.');
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Builder.php]: The first exception message you had was fine?

File: src/Illuminate/Database/Eloquent/Relations/BelongsToMany.php [hunk 6]
```diff
@@ -197,6 +198,9 @@ public function get($columns = ['*'])
      */
     public function paginate($perPage = null, $columns = ['*'], $pageName = 'page')
     {
+        if ($perPage < 0) {
+            throw new InvalidArgumentException('$perPage cannot be a negative value.');
+        }
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Relations/BelongsToMany.php]: missing newline

File: src/Illuminate/Database/Query/Builder.php [hunk 7]
```diff
@@ -1399,6 +1399,9 @@ protected function runSelect()
      */
     public function paginate($perPage = 15, $columns = ['*'], $pageName = 'page')
     {
+        if ($perPage < 0) {
+            throw new InvalidArgumentException('$perPage cannot be a negative value.');
+        }
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Builder.php]: missing newline

File: src/Illuminate/Database/Eloquent/Builder.php [hunk 8]
```diff
@@ -268,6 +269,10 @@ public function lists($column, $key = null)
      */
     public function paginate($perPage = null, $columns = ['*'], $pageName = 'page', $page = null)
     {
+        if ($perPage < 0) {
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Builder.php]: This is wrong anyway. Per page has to be greater than 0, not equal to or greater than 0.

**Comment by phpawy** [comment]
@GrahamCampbell 
added new lines
used the first error message
fixed comparison operator

What do you think?

**Comment by GrahamCampbell** [comment]
Please send a single commit and provide a useful exception message. Don't include variable names.
